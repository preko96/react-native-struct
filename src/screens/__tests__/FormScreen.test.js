import React from 'react'
import { FormScreen } from '../FormScreen/'
import { FormBuilder } from '../../resources/js/formBuilder'

import configureMockStore from 'redux-mock-store'
import thunkMiddleware from 'redux-thunk'
export const mockStore = configureMockStore([thunkMiddleware])
import { mount } from 'enzyme'

jest.mock('react-native-sqlite-storage', () => {
	const mockSQLite = {
		openDatabase: () => {
			return {
				transaction: () => {
					() => { return [] }
				}
			}
		}
	}
	return mockSQLite
})

jest.mock('../../resources/js/formBuilder', ()=>{
	const FormBuilder = () => {
		return {
			Reset: () => {
				return FormBuilder
			},
			Render: () => {
				return null
			},
			FromJSON: (json) => {
				return FormBuilder
			},
			Fetch: () => {
				return new Promise(resolve => {
					resolve([])
				})
			}
		}
	}

	return { FormBuilder }
})

jest.mock('../../resources/js/navigationHelper', () => {
	const navigationHelper = {
		getNavigationParams: () => {
			return
		}
	}

	return navigationHelper
})



it('renders without crashing', () => {
	const store = mockStore(
		{ 
			form: {json: ''}
		}
	)
	expect(shallow(<FormScreen store={store} />)).toMatchSnapshot()
})

it('renders without crashing mount', () => {
	jest.mock('react-native', () => require('react-native-mock-render/mock'), {virtual: true})
	const store = mockStore(
		{ 
			form: {json: ''}
		}
	)
	const wrapper = mount(<FormScreen store={store} />)
	wrapper.mount()
	
	expect(wrapper).toMatchSnapshot()
})

it('test load form function', async() => {
	jest.mock('react-native', () => require('react-native-mock-render/mock'), {virtual: true})

	const store = mockStore(
		{ 
			form: {json: '[{"type":"label","params":{"Value":"","Title":"Test Label","Style":{"fontSize":12,"fontColour":"#000","fontWeight":400},"Column":"label","Table":"testForm","FromDatabase":false,"PrimaryKeys":{}}}]'}
		}
	)
	const wrapper = mount(<FormScreen store={store}/>)
	wrapper.setProps({json: '[{"type":"label","params":{"Value":"","Title":"Test Label","Style":{"fontSize":12,"fontColour":"#000","fontWeight":400},"Column":"label","Table":"testForm","FromDatabase":false,"PrimaryKeys":{}}}]'})
	// await wrapper.instance().loadForm()
	
	expect(wrapper.state('Form')).toMatchSnapshot()
})