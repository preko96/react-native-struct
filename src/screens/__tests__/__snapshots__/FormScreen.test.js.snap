// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Styled(Container)>
  <Styled(Content) />
</Styled(Container)>
`;

exports[`renders without crashing mount 1`] = `
<FormScreen
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Styled(Container)>
    <Container
      style={
        Object {
          "flex": 1,
          "height": 1334,
        }
      }
    >
      <Component
        style={
          Object {
            "flex": 1,
            "height": 1334,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "height": 1334,
            }
          }
        >
          <Styled(Content)>
            <Content
              style={
                Object {
                  "backgroundColor": "transparent",
                  "flex": 1,
                }
              }
            >
              <_class
                automaticallyAdjustContentInsets={false}
                contentContainerStyle={
                  Array [
                    Object {
                      "padding": undefined,
                    },
                    undefined,
                  ]
                }
                enableAutomaticScroll={true}
                enableResetScrollToCoords={true}
                extraHeight={75}
                extraScrollHeight={0}
                keyboardOpeningTime={250}
                keyboardShouldPersistTaps="handled"
                resetScrollToCoords={
                  Object {
                    "x": 0,
                    "y": 0,
                  }
                }
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "flex": 1,
                  }
                }
                viewIsInsideTabBar={false}
              >
                <ScrollView
                  automaticallyAdjustContentInsets={false}
                  contentContainerStyle={
                    Array [
                      Object {
                        "padding": undefined,
                      },
                      undefined,
                    ]
                  }
                  contentInset={
                    Object {
                      "bottom": 0,
                    }
                  }
                  enableAutomaticScroll={true}
                  enableResetScrollToCoords={true}
                  extraHeight={75}
                  extraScrollHeight={0}
                  getScrollResponder={[Function]}
                  handleOnScroll={[Function]}
                  keyboardDismissMode="interactive"
                  keyboardOpeningTime={250}
                  keyboardShouldPersistTaps="handled"
                  keyboardSpace={0}
                  onScroll={[Function]}
                  resetKeyboardSpace={[Function]}
                  resetScrollToCoords={
                    Object {
                      "x": 0,
                      "y": 0,
                    }
                  }
                  scrollEventThrottle={1}
                  scrollForExtraHeightOnAndroid={[Function]}
                  scrollToEnd={[Function]}
                  scrollToFocusedInput={[Function]}
                  scrollToPosition={[Function]}
                  showsVerticalScrollIndicator={true}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "flex": 1,
                    }
                  }
                  viewIsInsideTabBar={false}
                >
                  <_class
                    automaticallyAdjustContentInsets={false}
                    contentContainerStyle={
                      Array [
                        Object {
                          "padding": undefined,
                        },
                        undefined,
                      ]
                    }
                    contentInset={
                      Object {
                        "bottom": 0,
                      }
                    }
                    enableAutomaticScroll={true}
                    enableResetScrollToCoords={true}
                    extraHeight={75}
                    extraScrollHeight={0}
                    getScrollResponder={[Function]}
                    handleOnScroll={[Function]}
                    keyboardDismissMode="interactive"
                    keyboardOpeningTime={250}
                    keyboardShouldPersistTaps="handled"
                    keyboardSpace={0}
                    onScroll={[Function]}
                    resetKeyboardSpace={[Function]}
                    resetScrollToCoords={
                      Object {
                        "x": 0,
                        "y": 0,
                      }
                    }
                    scrollEventThrottle={1}
                    scrollForExtraHeightOnAndroid={[Function]}
                    scrollToEnd={[Function]}
                    scrollToFocusedInput={[Function]}
                    scrollToPosition={[Function]}
                    showsVerticalScrollIndicator={true}
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "flex": 1,
                      }
                    }
                    viewIsInsideTabBar={false}
                  >
                    <RCTScrollView
                      automaticallyAdjustContentInsets={false}
                      contentContainerStyle={
                        Array [
                          Object {
                            "padding": undefined,
                          },
                          undefined,
                        ]
                      }
                      contentInset={
                        Object {
                          "bottom": 0,
                        }
                      }
                      enableAutomaticScroll={true}
                      enableResetScrollToCoords={true}
                      extraHeight={75}
                      extraScrollHeight={0}
                      getScrollResponder={[Function]}
                      handleOnScroll={[Function]}
                      keyboardDismissMode="interactive"
                      keyboardOpeningTime={250}
                      keyboardShouldPersistTaps="handled"
                      keyboardSpace={0}
                      onScroll={[Function]}
                      resetKeyboardSpace={[Function]}
                      resetScrollToCoords={
                        Object {
                          "x": 0,
                          "y": 0,
                        }
                      }
                      scrollEventThrottle={1}
                      scrollForExtraHeightOnAndroid={[Function]}
                      scrollToEnd={[Function]}
                      scrollToFocusedInput={[Function]}
                      scrollToPosition={[Function]}
                      showsVerticalScrollIndicator={true}
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "flex": 1,
                        }
                      }
                      viewIsInsideTabBar={false}
                    >
                      <Component>
                        <View />
                      </Component>
                    </RCTScrollView>
                  </_class>
                </ScrollView>
              </_class>
            </Content>
          </Styled(Content)>
        </View>
      </Component>
    </Container>
  </Styled(Container)>
</FormScreen>
`;

exports[`test load form function 1`] = `
Object {
  "Fetch": [Function],
  "FromJSON": [Function],
  "Render": [Function],
  "Reset": [Function],
}
`;
